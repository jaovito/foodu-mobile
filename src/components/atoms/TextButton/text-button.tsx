import React from 'react';
import {TextButtonProps} from './text-button.props';
import {Container, Text} from './text-button.styles';

export const TextButton = ({
  children,
  textProps,
  ...props
}: TextButtonProps) => {
  return (
    <Container activeOpacity={0.7} {...props}>
      <Text {...textProps}>{children}</Text>
    </Container>
  );
};
