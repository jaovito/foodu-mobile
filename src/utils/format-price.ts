import {LanguagePaths} from '@translations/enumns/LanguagePaths';
import {translate} from '@translations/index';

const defaultValues = {
  language: 'en-US',
  currency: 'USD',
};

function getDefaultValues() {
  if (translate(LanguagePaths.GENERAL_LANGUAGE) === 'general.language') {
    return defaultValues;
  }

  return {
    language: translate(LanguagePaths.GENERAL_LANGUAGE),
    currency: translate(LanguagePaths.GENERAL_CURRENCY),
  };
}

export function formatPrice(price: number) {
  return price.toLocaleString(getDefaultValues().language, {
    style: 'currency',
    currency: getDefaultValues().currency,
  });
}
