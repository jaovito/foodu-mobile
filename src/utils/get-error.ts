import {AxiosError} from 'axios';

export function getError(err: Error | unknown): AxiosError {
  if (err instanceof AxiosError) {
    return err;
  }

  return err as AxiosError;
}
