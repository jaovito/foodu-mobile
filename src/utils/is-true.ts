/**
 *
 * Convert booleans values, considering the cases that I understand how true and false.
 * Exemple: Empty arrays and objects, we consider false.
 *
 * @param  param - Can be any value
 *
 * @returns if `value` it isn't an empty array, returns true
 * @returns if `value` it isn't an empty object, returns true
 * @returns if `value` it isn't an empty string, returns true
 * @returns if `value` it isn't null, returns true
 * @returns if `value` it's a non-zero number, returns true
 */
export const isTrue = (param: any): boolean => {
  if (typeof param === 'object') {
    if (param === null) {
      return false;
    } else if (param instanceof Array) {
      return param.length === 0 ? false : true;
    } else {
      return Object.entries(param).length === 0 ? false : true;
    }
  }

  return !!param;
};
