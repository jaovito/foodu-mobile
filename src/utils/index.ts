import {isTrue} from './is-true';

export {isTrue};
