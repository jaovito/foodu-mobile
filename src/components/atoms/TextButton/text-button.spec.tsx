import {fireEvent, render} from '@test/utils/test-utils';
import {TextButton} from './text-button';

describe('<TextButton />', () => {
  const title = 'Save';
  it('should be defined', () => {
    const button = render(<TextButton>{title}</TextButton>);

    expect(button).toBeDefined();
  });

  it('should be able to find the text inside the button', async () => {
    const {findByText} = render(<TextButton>{title}</TextButton>);

    const text = await findByText(title);

    expect(text).toBeDefined();
    expect(text).toBeTruthy();
  });

  it('should be able to click on a text button', async () => {
    const onPress = jest.fn();
    const {findByText} = render(
      <TextButton onPress={onPress}>{title}</TextButton>,
    );

    const text = await findByText(title);
    fireEvent.press(text);

    expect(onPress).toHaveBeenCalled();
  });

  it('should be able to match with snapshot', async () => {
    const onPress = jest.fn();
    const button = render(
      <TextButton onPress={onPress}>{title}</TextButton>,
    ).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should be able to match with snapshot without onPress', async () => {
    const button = render(<TextButton>{title}</TextButton>).toJSON();

    expect(button).toMatchSnapshot();
  });
});
