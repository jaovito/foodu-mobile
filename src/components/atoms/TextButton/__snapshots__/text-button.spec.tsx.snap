// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextButton /> should be able to match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {},
      ]
    }
  >
    Save
  </Text>
</View>
`;

exports[`<TextButton /> should be able to match with snapshot without onPress 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {},
      ]
    }
  >
    Save
  </Text>
</View>
`;
